scriptmodules/supplementary/skyscraper.sh

```diff
@@ -199,44 +203,32 @@ function configure_skyscraper() {
 function _init_config_skyscraper() {
     local scraper_conf_dir="$configdir/all/skyscraper"
 
-    # Make sure the `artwork.xml` and other conf file(s) are present, but don't overwrite them on upgrades
-    local f_conf
-    for f_conf in artwork.xml aliasMap.csv; do
-        if [[ -f "$scraper_conf_dir/$f_conf" ]]; then
-            cp -f "$md_inst/$f_conf" "$scraper_conf_dir/$f_conf.default"
-        else
-            cp "$md_inst/$f_conf" "$scraper_conf_dir"
-        fi
-    done
-
     # If we don't have a previous config.ini file, copy the example one
-    if [[ ! -f "$scraper_conf_dir/config.ini" ]]; then
-        cp "$md_inst/config.ini.example" "$scraper_conf_dir/config.ini"
-        sed -i 's/\[esgamelist\]/[esgamelist]\ncacheScreenshots="false"/' "$scraper_conf_dir/config.ini"
-    fi
+    [[ ! -f "$scraper_conf_dir/config.ini" ]] && cp "$md_inst/config.ini.example" "$scraper_conf_dir/config.ini"
 
     # Try to find the rest of the necessary files from the qmake build file
-    # They should be listed in the `unix:examples.file` configuration line
-    if [[ $(grep unix:examples.files "$md_build/skyscraper.pro" 2>/dev/null | cut -d= -f2-) ]]; then
-        local files=$(grep unix:examples.files "$md_build/skyscraper.pro" | cut -d= -f2-)
-        local file
-
-        for file in $files; do
-            # Copy the files to the configuration folder. Skip config.ini, artwork.xml and aliasMap.csv
-            if [[ $file != "artwork.xml" && $file != "config.ini" && $file != "aliasMap.csv" ]]; then
-                cp -f "$md_build/$file" "$scraper_conf_dir"
-            fi
-        done
-    else
-        # Fallback to the known resource files list
-        cp -f "$md_inst/artwork.xml.example"* "$scraper_conf_dir"
-
-        # Copy resources and readme
-        local resource_file
-        for resource_file in README.md mameMap.csv tgdb_developers.json tgdb_publishers.json hints.txt; do
-            cp -f "$md_inst/$resource_file" "$scraper_conf_dir"
-        done
+    # They should be listed in the `unix:examples.files` configuration line
+    iniConfig "=" "" "$md_build/skyscraper.pro"
+    iniGet "unix:examples.files"
+    local files="$ini_value"
+    if [[ -z "$files" ]]; then
+        files="hints.xml artwork.xml artwork.xml.example1 artwork.xml.example2 artwork.xml.example3"
+        files+=" artwork.xml.example4 aliasMap.csv mameMap.csv tgdb_developers.json tgdb_publishers.json"
+        files+=" platforms.json screenscraper.json mobygames.json"
     fi
+    local file
+    for file in $files; do
+        case "$file" in
+            config.ini|README.md|docs/ARTWORK.md)
```

Comment:
Please remove the `.md` document files, they shouldn't be copied into the user's confi folder.

Reply:
Naming those files at this point I think is necessary in order to ensure that they are *not* copied to the config folder, even if they might be named in `$files` (as during a source install, when `$files` [is read from](https://github.com/RetroPie/RetroPie-Setup/blob/0ab5aa17835bd14a9f82e4499f878cfa515ab5e5/scriptmodules/supplementary/skyscraper.sh#L211-L213) the [qmake build file skyscraper.pro](https://github.com/Gemba/skyscraper/blob/e8969834a6f00fbd06b64112b7363e0606ad4c29/skyscraper.pro#L14) and not from the [fallback list that's written in the scriptmodule](https://github.com/RetroPie/RetroPie-Setup/blob/0ab5aa17835bd14a9f82e4499f878cfa515ab5e5/scriptmodules/supplementary/skyscraper.sh#L214-L217).) Or else @Gemba can remove them from **unix:examples.files** so they're not included in `$files` but if I remove the filenames from this line in the scriptmodule now, while they're still named on that line in the build file, then they *will be copied* to the config folder at [lines 227-228](https://github.com/RetroPie/RetroPie-Setup/blob/0ab5aa17835bd14a9f82e4499f878cfa515ab5e5/scriptmodules/supplementary/skyscraper.sh#L227-L228). Naming them here at [L222-L223](https://github.com/RetroPie/RetroPie-Setup/blob/0ab5aa17835bd14a9f82e4499f878cfa515ab5e5/scriptmodules/supplementary/skyscraper.sh#L222-L223) says "do nothing" with them and ensures that they are *not* copied, instead.

Reply:
Then I think I'd prefer to explicitely list the files based on their copy mode (overwrite or not) and get rid of the `unix:example.files` altogether. Either way is fine.

Reply:
> explicitely list the files based on their copy mode (overwrite or not) and get rid of the `unix:example.files` altogether.

The downside here being, that if new resource files are ever required (such as the new JSON files, this time), then a source install won't work until the scriptmodule is updated to include them. (Nor the binary, supposing it is just built from source and then archived.)

For the record, the current RP scriptmodule featuring muldjord already copies both of these files in a source install, and explicitly includes README.md in the binary also.

Reply:
Yes, that was certainly the original intention, but in practice the file list doesn't change that often and the required files need to be added to `install_` anyway to accomodate binary installs.
I think it's better to be explicit, makes it easier to understand what files are neeed and their purpose. Thanks for the changes.